src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleSmartEat.kt

```diff
@@ -0,0 +1,194 @@
+package net.ccbluex.liquidbounce.features.module.modules.combat
+
+import net.ccbluex.liquidbounce.config.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.events.OverlayRenderEvent
+import net.ccbluex.liquidbounce.event.events.PlayerInteractedItem
+import net.ccbluex.liquidbounce.event.handler
+import net.ccbluex.liquidbounce.event.repeatable
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.Module
+import net.ccbluex.liquidbounce.render.renderEnvironmentForGUI
+import net.ccbluex.liquidbounce.utils.client.SilentHotbar
+import net.ccbluex.liquidbounce.utils.client.chat
+import net.ccbluex.liquidbounce.utils.item.Hotbar
+import net.minecraft.client.gui.DrawContext
+import net.minecraft.client.option.KeyBinding
+import net.minecraft.entity.effect.StatusEffects
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.item.ItemStack
+import net.minecraft.item.Items
+import net.minecraft.potion.PotionUtil
+import net.minecraft.util.ActionResult
+import net.minecraft.util.Hand
+import net.minecraft.util.Identifier
+import net.minecraft.util.UseAction
+import kotlin.math.max
+
+/**
+ * AutoClicker module
+ *
+ * Clicks automatically when holding down a mouse button.
+ */
+
+object ModuleSmartEat : Module("SmartEat", Category.PLAYER) {
+    private val HOTBAR_OFFHAND_LEFT_TEXTURE = Identifier("hud/hotbar_offhand_left")
+
+
+
+    private val swapBackDelay by int("SwapBackDelay", 5, 1..20)
+
+    private val preferGappleHealth by float("PreferGappleHealth", 9f, 0f..20f)
+    private val preferNotchAppleHealth by float("PreferNotchAppleHealth", 2f, 0f..20f)
+    private val preferHealthPotHealth by float("PreferHealthPotHealth", 12f, 0f..20f)
+
+    private var prefersGapples = false
+    private var prefersNotchApple = false
+    private var prefersHealthPot = false
+    private val food: Pair<Int, ItemStack>?
+        get() = Hotbar.findBestItem(0) { _, itemStack ->
+            val item = itemStack.item
+
+            if (prefersHealthPot && item == Items.POTION) {
+                val hasHealthEffect =
+                    PotionUtil.getPotionEffects(itemStack).any {
+                        it.effectType == StatusEffects.INSTANT_HEALTH
+                    }
+
+                if (hasHealthEffect)
+                    return@findBestItem 100 - preferHealthPotHealth.toInt()
+            }
+
+            val foodComp = item.foodComponent ?: return@findBestItem -1
+
+            val hasHungerEffect = foodComp.statusEffects.any { it.first.effectType == StatusEffects.HUNGER }
+            if (hasHungerEffect)
+                return@findBestItem 0
+
+            if(prefersGapples && item == Items.GOLDEN_APPLE)
+                return@findBestItem 100 - preferGappleHealth.toInt()
+
+            if (prefersNotchApple && item == Items.ENCHANTED_GOLDEN_APPLE)
+                return@findBestItem 100 - preferNotchAppleHealth.toInt()
+
+            foodComp.hunger
+        }
```

Comment:
1. This should not be a property, rather a function (`findFoodSlotsInHotbar` or something)
2. Using ints to refer to item slots is deprecated since it causes uneccessary bugs. Use `ItemSlot`s instead: https://github.com/CCBlueX/LiquidBounce/blob/63d9d141e73d25e2bd18ccdf81f93e7414753a39/src/main/kotlin/net/ccbluex/liquidbounce/utils/item/InventoryUtils.kt#L48 Maybe add a version of `ALL_SLOTS_IN_INVENTORY` for hotbar like `HOTBAR_SLOTS_IN_INVENTORY`.
3. The comparision part should be it's own function like `getFoodItemScore(ItemStack)`, so the code become more intuitive

Reply:
> This should not be a property, rather a function (findFoodSlotsInHotbar or something)

No problem, but i actually got inspired from this code: https://github.com/CCBlueX/LiquidBounce/blob/61099831d1a31961f66a241d0ae51d4ba7b67f6b/src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/nofall/modes/MLG.kt#L57-L62